presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -57,6 +63,27 @@ public void setup()
     {
         connection = createConnection();
         statement = connection.createStatement();
+        synchronized (this) {
+            determinePrestoVersion();
+        }
```

Comment:
The optional is not for sake of setting it up. ONly because the version is not available always. E.g if you run test from intellij you will not get version information.
I can make it supplier. But it will still need to either return `Optional` or `"unknown"`

Reply:
> yet still the version can be not parsable.

i think we should fail loud in such case -- https://github.com/prestosql/presto/pull/4998#discussion_r484433758